### ISSUE TITLE
[package] xorg/system: some system include directories are added

### ISSUE THREAD

**Issue by madebr** [conan-io/conan-center-index#6880]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **xorg/system**
  * Operating System+version: **Fedora Linux 34**
  * Compiler+version: **GCC 11.2**
  * Conan version: **conan 1.39.0**
  * Python version: **Python 3.9.6**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=11
compiler.libcxx=libstdc++11
build_type=Release
[options]
[conf]
[build_requires]
[env]
CC=/usr/bin/gcc
CXX=/usr/bin/g++
```

### Steps to reproduce (Include if Applicable)
```
conan create recipes/xorg/all xorg/system@
cat recipes/xorg/all/test_package/build/220a4431a5a2a0f3bef67db079657b4ddcad22b7/conanbuildinfo.txt
```

The `[includedirs_xorg]` sections is possibibly problematic:
```
[includedirs_xorg]
/usr/include/freetype2
/usr/include/libpng16
/usr/include/harfbuzz
/usr/include/glib-2.0
/usr/lib64/glib-2.0/include
/usr/include/sysprof-4
/usr/include/libxml2
```

**Comment by ericLemanissier** [comment]
yes, xorg-dev brings libxft-dev which brings libfreetype6-dev and libfontconfig1-dev. I'm not sure where libpng, harfbuzz glib etc. come from. This is far from ideal, but I currently have no idea how we could replace these system packages with conan packages in this case

**Comment by madebr** [comment]
Those are dependencies of dependencies.
```
$ pkg-config --cflags-only-I xft
-I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libxml2
```

```
$ cat /usr/lib64/pkgconfig/xft.pc
 cat /usr/lib64/pkgconfig/xft.pc
prefix=/usr
exec_prefix=/usr
libdir=/usr/lib64
includedir=/usr/include

Name: Xft
Description: X FreeType library
Version: 2.3.3
Requires: xproto
Requires.private: xrender, fontconfig, freetype2
Cflags: -I${includedir}
Libs: -L${libdir} -lXft
```
So it looks like it pulls in the private requires.
Private requires are those that are needed when a static library is linked against.

I don't understand how my system pkg-config returns the same result for `--static` and `--shared`:
```
[maarten@fedora ~]$ pkg-config --static --cflags-only-I xft
-I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libxml2 
[maarten@fedora ~]$ pkg-config --shared --cflags-only-I xft
-I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/libxml2 
```

**Comment by ericLemanissier** [comment]
I don't know this `--shared` command line argument, but according to https://people.freedesktop.org/~dbn/pkg-config-guide.html: `pkg-config needs to output both sets of link flags in this case to ensure that the statically linked application will find all the necessary symbols. On the other hand, it will always output all the Cflags.`, and https://linux.die.net/man/1/pkg-config : `--static: output libraries suitable for static linking`, and `pkg-config --help` : `--static                                output linker flags for static linking`
bottom line: it seems that compilation flags of private dependencies are always returned.

**Comment by madebr** [comment]
Yup, you're right about pkg-config outputting all the cflags and `--static`/`--shared` is only changing the link-releated flags.
